Marlin/src/lcd/ultralcd.cpp

```diff
@@ -3998,6 +4010,186 @@ void kill_screen(const char* lcd_msg) {
+        SERIAL_ECHOLNPAIR("led_old_white: ", led_old_white);
+      #endif
+      #if ENABLED(NEOPIXEL_LED)
+        SERIAL_ECHOLNPAIR("led_old_brightness: ", led_old_brightness);
+      #endif
+      SERIAL_ECHOLN("===== AFTER UPDATE =====");
+
+      set_led_color(led_red_intensity, led_green_intensity, led_blue_intensity
+      #if ENABLED(RGBW_LED)
+        , led_white_intensity
+      #endif
+      #if ENABLED(NEOPIXEL_LED)
+        , 0, led_brightness
+      #endif
+      );
+    }
+
+    void set_user_color() {
+
+      // These debug lines will be removed after all is good.
+      SERIAL_ECHOLN("====== BEFORE ======");
+      SERIAL_ECHOLNPAIR("led_old_red: ", led_old_red);
+      SERIAL_ECHOLNPAIR("led_old_green: ", led_old_green);
+      SERIAL_ECHOLNPAIR("led_old_blue: ", led_old_blue);
+      #if ENABLED(RGBW_LED)
+        SERIAL_ECHOLNPAIR("led_old_white: ", led_old_white);
+      #endif
+      #if ENABLED(NEOPIXEL_LED)
+        SERIAL_ECHOLNPAIR("led_old_brightness: ", led_old_brightness);
+      #endif
+      SERIAL_ECHOLN("====== BEFORE ======");
+
+      if (led_save_color)
+        update_leds();
+      else
+        set_led_color(LED_USER_PRESET_RED, LED_USER_PRESET_GREEN, LED_USER_PRESET_BLUE
+        #if ENABLED(RGBW_LED)
+          , LED_USER_PRESET_WHITE
+        #endif
+        #if ENABLED(NEOPIXEL_LED)
+          , 0, LED_USER_PRESET_INTENSITY
+        #endif
+        );
+
+        // These debug lines will be removed after all is good.
+      SERIAL_ECHOLN("===== AFTER NO SAVE =====");
+      SERIAL_ECHOLNPAIR("led_old_red: ", led_old_red);
+      SERIAL_ECHOLNPAIR("led_old_green: ", led_old_green);
+      SERIAL_ECHOLNPAIR("led_old_blue: ", led_old_blue);
+      #if ENABLED(RGBW_LED)
+        SERIAL_ECHOLNPAIR("led_old_white: ", led_old_white);
+      #endif
+      #if ENABLED(NEOPIXEL_LED)
+        SERIAL_ECHOLNPAIR("led_old_brightness: ", led_old_brightness);
+      #endif
+      SERIAL_ECHOLN("===== AFTER NO SAVE =====");
+    }
+
+    void led_save_user_color() {
+      led_save_color = true;
+      set_user_color();
+    }
+
+    void led_restore_default() {
+      led_save_color = false;
+      set_user_color();
+    }
+
+    void set_leds_off() {
+      set_led_color(0, 0, 0
+      #if ENABLED(RGBW) || ENABLED(NEOPIXEL_LED)
+      , 0
+      #endif
+      );
+    }
+
+    void lcd_led_presets_menu() {
+      START_MENU();
+      MENU_BACK(MSG_LED_CONTROL);
+      MENU_ITEM(gcode, MSG_LED_ON MSG_RED MSG_LIGHTS, PSTR("M150 R255"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_ORANGE MSG_LIGHTS, PSTR("M150 R255 U60"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_YELLOW MSG_LIGHTS, PSTR("M150 R255 U255"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_GREEN MSG_LIGHTS, PSTR("M150 U255"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_BLUE MSG_LIGHTS, PSTR("M150 B255"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_PURPLE MSG_LIGHTS, PSTR("M150 R255 B255"));
+      MENU_ITEM(gcode, MSG_LED_ON MSG_WHITE MSG_LIGHTS, PSTR("M150 W255"));
```

Comment:
G-code strings may possibly be larger than code. ("255" is 3 bytes vs 1-byte `0xFF`) Check to see whether code to set LED color via `set_led_color()` compiles smaller…
```cpp
void lcd_led_set_red() { set_led_color(255, 0, 0); }
. . .
MENU_ITEM(function, MSG_LED_ON MSG_RED MSG_LIGHTS, lcd_led_red);
```

Reply:
Working on this one...